src/Serialization/ToricGeometry.jl

```diff
@@ -93,3 +93,29 @@ function load_object(s::DeserializerState, ::Type{ToricDivisorClass}, tv::Normal
   pmdiv = Polymake._lookup_multi(pm_object(tv), "DIVISOR", index-1)
   return toric_divisor_class(ToricDivisor(pmdiv, tv, coeffs))
 end
+
+################################################################################
+# Cohomology classes on toric varieties
+@register_serialization_type CohomologyClass uses_params
+
+function save_type_params(s::SerializerState, obj::CohomologyClass)
+  save_data_dict(s) do
+    save_object(s, encode_type(CohomologyClass), :name)
+    save_typed_object(s, obj.v, :params)
+  end
+end
+
+function load_type_params(s::DeserializerState, ::Type{<:CohomologyClass})
+  return load_typed_object(s)
+end
+
+function save_object(s::SerializerState, cc::CohomologyClass)
+  save_data_dict(s) do
+    save_typed_object(s, polynomial(cc).f, :polynomial)
```

Comment:
Aside, I tried `save_typed_object(s, polynomial(cc), :polynomial)` first. Unless I am mistaken, serialization for `MPolyQuoRingElem` is not yet supported, correct?

Reply:
I think @HechtiDerLachs added support for this? Maybe it hasn't been merged?

Reply:
Ok. But do we really need this here?

The current version of the code tries to serialize a polynomial in the base ring (rather than the quotient ring). In theory, this is sufficient to serialize a (toric) cohomology class. But the loading of said polynomial does not work (yet) - reasons unclear/unknown to me.